pkg/odo/cli/pipelines/service/add.go

```diff
@@ -70,6 +82,8 @@ func newCmdAdd(name, fullName string) *cobra.Command {
 	cmd.Flags().StringVar(&o.appName, "app-name", "", "the name of the application where the service will be added")
 	cmd.Flags().StringVar(&o.serviceName, "service-name", "", "the name of the service to be added")
 	cmd.Flags().StringVar(&o.envName, "env-name", "", "the name of the environment where the service will be added")
+	cmd.Flags().StringVar(&o.imageRepo, "image-repo", "", "used to push built images")
+	cmd.Flags().StringVar(&o.internalRegistryHostname, "internal-registry-hostname", "image-registry.openshift-image-registry.svc:5000", "internal image registry hostname")
 	cmd.Flags().StringVar(&o.manifest, "manifest", "pipelines.yaml", "path to manifest file")
 
 	// required flags
```

Comment:
We use `dockercjgjson` flag for an external registry. Can we make it a mandatory flag only if an external registry is used?